com.unity.shadergraph/Editor/GraphUI/EditorCommon/Preview/PreviewManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using Editor.GraphUI.Utilities;
+using UnityEditor.ShaderGraph.GraphDelta;
+using UnityEditor.ShaderGraph.GraphUI.DataModel;
+using UnityEngine.GraphToolsFoundation.Overdrive;
+using UnityEngine;
+using UnityEngine.Rendering;
+
+namespace UnityEditor.ShaderGraph.GraphUI.EditorCommon.Preview
+{
+    using PreviewRenderMode = HeadlessPreviewManager.PreviewRenderMode;
+
+    public class PreviewManager
+    {
+        HeadlessPreviewManager m_PreviewHandlerInstance;
+
+        HashSet<string> m_DirtyNodes;
+
+        ShaderGraphModel m_GraphModel;
+
+        Dictionary<string, SerializableGUID> m_NodeLookupTable;
+
+        public PreviewManager(ShaderGraphModel graphModel)
+        {
+            m_GraphModel = graphModel;
+
+            m_PreviewHandlerInstance = new HeadlessPreviewManager();
+
+            m_PreviewHandlerInstance.SetActiveGraph(m_GraphModel.GraphHandler);
+            m_PreviewHandlerInstance.SetActiveRegistry(m_GraphModel.RegistryInstance);
+
+            m_DirtyNodes = new HashSet<string>();
+            m_NodeLookupTable = new Dictionary<string, SerializableGUID>();
+
+            // Initialize preview data for any nodes that exist on graph load
+            foreach (var nodeModel in graphModel.NodeModels)
+            {
+                if(nodeModel is GraphDataNodeModel graphDataNodeModel)
+                    OnNodeAdded(graphDataNodeModel.graphDataName, graphDataNodeModel.Guid);
+            }
+        }
+
+        public void Update()
```

Comment:
I would like to see about getting this behavior out of an update function, but this is fine for now.